extension-style/src/main/java/com/mapbox/maps/extension/style/projection/generated/ProjectionExt.kt

```diff
@@ -0,0 +1,66 @@
+// This file is generated.
+@file:JvmName("StyleProjectionUtils")
+
+package com.mapbox.maps.extension.style.projection.generated
+
+import com.mapbox.maps.StylePropertyValueKind
+import com.mapbox.maps.extension.style.StyleInterface
+import com.mapbox.maps.extension.style.layers.properties.generated.ProjectionName
+import com.mapbox.maps.extension.style.utils.unwrapToAny
+
+/**
+ * Extension function to get the projection name provided by the Style Extension.
+ *
+ * @return projection [ProjectionName] that is currently applied to the map
+ */
+fun StyleInterface.getProjectionName(): ProjectionName {
+  getStyleProjectionProperty("name").apply {
+    return if (kind == StylePropertyValueKind.UNDEFINED) {
+      ProjectionName.MERCATOR
+    } else {
+      val projectionAsString = value.unwrapToAny() as String
+      return ProjectionName.valueOf(projectionAsString.uppercase())
+    }
+  }
+}
+
+/**
+ * Extension function to get the projection provided by the Style Extension.
+ *
+ * @return projection [Projection] that is currently applied to the map
+ */
+fun StyleInterface.getProjection(): Projection {
+  val projection = Projection()
+  getStyleProjectionProperty("name").apply {
+    if (kind == StylePropertyValueKind.UNDEFINED) {
+      projection.name(ProjectionName.MERCATOR)
+    } else {
+      val projectionAsString = value.unwrapToAny() as String
+      projection.name(ProjectionName.valueOf(projectionAsString.uppercase()))
+    }
+  }
+  projection.bindTo(this)
+  return projection
+}
+
+/**
+ * Extension function to set the Projection by name provided by the Style Extension to the Style.
+ *
+ * @param projectionName The projection [ProjectionName] to be set.
+ */
+fun StyleInterface.setProjectionName(projectionName: ProjectionName) {
```

Comment:
propose to remove this extension

Reply:
I think as we have projection DSL, it could be simplified as 
```
style.setProjection(
  projection {
    name(ProjectionName.GLOBE)
  } 
)
```

I think it's a more aligned API interface with terrain and light, and it's also future-proof.